checkpoint/orbax/checkpoint/_src/serialization/replica_slices_test.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 The Orbax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+from absl.testing import parameterized
+import jax
+import numpy as np
+from orbax.checkpoint._src.serialization.replica_slices import get_replica_slices, transfer_arrays_to_host
+
+
+def make_multi_device_array(shape, spec):
+  key = jax.random.PRNGKey(0)
+  x = jax.random.normal(jax.random.PRNGKey(0), shape)
+  mesh = jax.sharding.Mesh(jax.devices(), ('x',))
+  sharding = jax.sharding.NamedSharding(mesh, spec)
+  return jax.device_put(x, sharding)
+
+
+def is_pow_of_two(n):
+  while n > 1:
+    n, rem = divmod(n, 2)
+    if rem == 1:
+      return False
+  return True
+
+
+class ReplicaSlicesTest(parameterized.TestCase):
+
+  def setUp(self):
+    super().setUp()
+    num_devices = len(jax.devices())
+    assert num_devices >= 2
+    assert is_pow_of_two(num_devices)
+
+  def test_get_replica_slices_single_replica(self):
```

Comment:
Would be good to add new cases, or parameterize the existing ones, to include tests for arrays that are not fully replicated.

Reply:
Added a second variant of the test that operates on a partially-replicated array (first dim partitioned).